src/PhpSpec/Console/ContainerAssembler.php

```diff
@@ -320,11 +329,17 @@ private function setupGenerators(IndexedServiceContainer $container)
         }, ['code_generator.generators']);
 
         $container->define('code_generator.generators.private_constructor', function (IndexedServiceContainer $c) {
-            return new CodeGenerator\Generator\PrivateConstructorGenerator(
-                $c->get('console.io'),
-                $c->get('code_generator.templates'),
-                $c->get('util.filesystem'),
-                $c->get('code_generator.writers.tokenized')
+            return new CodeGenerator\Generator\OneTimeGenerator(
```

Comment:
Please define these as services as well, so that extensions can add their own decorators